cql3/statements/ks_prop_defs.cc

```diff
@@ -155,6 +155,9 @@ std::optional<unsigned> ks_prop_defs::get_initial_tablets(std::optional<unsigned
         if (enabled == "true") {
             // nothing
         } else if (enabled == "false") {
+            if (enforce_tablets) {
+                throw exceptions::configuration_exception("Cannot disable tablets for keyspace since tablets are enforced using the `enforce_tablets` config option.");
+            }
```

Comment:
This creates a weird effect. Consider `--enforce-tablets=true` and user is running `create keyspace foo with replication { class: simple }`. In that case it will be created without tablets. Now user runs `create keyspace foo with replication { class: simple } and tablets { enabled: false }`. Now an exception will be thrown

Reply:
I'll add a test for that.
We should enforce tablets and reject SimpleStrategy, throwing an "invalid" error.

Reply:
Not only SimpleStrategy, I suppose, but any strategy that is not tablets-aware

Reply:
Added unit tests to verify the error in v2